dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityAttributeObjectBundleHook.java

```diff
@@ -74,25 +88,39 @@ public void validate(
       }
     }
 
-    if (attr.getPreferredSearchOperator() != null) {
-      if (!getTrackerOperators().contains(attr.getPreferredSearchOperator())) {
-        addReports.accept(
-            new ErrorReport(
-                TrackedEntityAttribute.class,
-                ErrorCode.E4081,
-                attr.getPreferredSearchOperator(),
-                getTrackerOperators()));
-      }
+    Collection<QueryOperator> providedNonBlockableOperators =
+        intersection(NON_BLOCKABLE_OPERATORS, attr.getBlockedSearchOperators());
 
-      if (attr.getBlockedSearchOperators() != null
-          && attr.getBlockedSearchOperators().contains(attr.getPreferredSearchOperator())) {
-        addReports.accept(
-            new ErrorReport(
-                TrackedEntityAttribute.class,
-                ErrorCode.E4082,
-                attr.getPreferredSearchOperator(),
-                attr.getUid()));
-      }
+    if (!providedNonBlockableOperators.isEmpty()) {
+      addReports.accept(
+          new ErrorReport(
+              TrackedEntityAttribute.class,
+              ErrorCode.E4088,
+              providedNonBlockableOperators,
+              NON_BLOCKABLE_OPERATORS));
+    }
+
+    if (attr.getPreferredSearchOperator() == null) {
+      return;
+    }
+
+    if (!getTrackerOperators().contains(attr.getPreferredSearchOperator())) {
+      addReports.accept(
+          new ErrorReport(
+              TrackedEntityAttribute.class,
+              ErrorCode.E4081,
+              attr.getPreferredSearchOperator(),
+              getTrackerOperators()));
+    }
+
+    if (providedNonBlockableOperators.isEmpty()
```

Comment:
can we remove `providedNonBlockableOperators.isEmpty()` here as E4088 and E4082 are independent errors, no?